gutenberg.php

```diff
@@ -159,3 +159,25 @@ function gutenberg_rest_nonce() {
 	exit( wp_create_nonce( 'wp_rest' ) );
 }
 add_action( 'wp_ajax_gutenberg_rest_nonce', 'gutenberg_rest_nonce' );
+
+/**
+ * Prints TinyMCE scripts when we're outside of the block editor.
+ * Otherwise, use the default TinyMCE script printing behavior.
+ *
+ * @since 7.9.1
+ */
+function gutenberg_print_tinymce_scripts() {
+	$current_screen = get_current_screen();
+	if ( ! $current_screen->is_block_editor() ) {
+		// maybe also need to check a setting/feature flag? Put this behind phase 2?
+		wp_print_tinymce_scripts();
+	} else {
+		echo "<!-- Skipping TinyMCE in favor of loading async -->";
+		echo "<script type='text/javascript'>\n" .
+			"window.wpMceTranslation = function() {\n" .
```

Comment:
Shouldn't it be part of the REST API response in this async scenario?

I don't think it can be as simple as that for the block editor. Plugins still can register [Meta Boxes](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/backward-compatibility/meta-box.md) that use TinyMCE heavily.

Reply:
Yeah. Definitely an open question to me... I guess we can create a filter? The issue is the endpoint returns just the URI for the dependency, not the dependency script itself, so we can’t just concatenate the translation bit to it.

An alternative to this is to just have a `tinymce-i18n` endpoint to just return the json needed, the async `onLoaded` prop is suitable for that.